terms/hyperparameters_nm.tex

```diff
@@ -0,0 +1,70 @@
+\chapter[Hyperparameters]{Hyperparameters}
+\label{ch:Hyperparameters}
+
+As stated by Li \textit{et al.} in \cite{Li2017}, "performance of machine learning algorithms depends critically on identifying a good set of hyperparameters". Hyperparameter tuning is a topic often discussed in literature describing any combination of machine learning algorithms. Adequate values set for hyperparameters will massively influence training of a model, be it in terms of processing time, as well as the actual robustness of the model in question. 
+
+\section{Introduction}\label{sec:page-layout}
+\subsection{Defining Hyperparameters}\label{sec:headings}\index{headings}
+Hyperparameters are variables which are initialised before a machine learning model undergoes training. The general representation is as a vector, with common examples of hyperparameters including the learning rate of a model, regularization coefficients, and kernel parameters. Considering the learning rate, an improperly set value will greatly influence the training potential of a regression model or neural network. Extremely high learning rate values may render a system volatile, causing weight/parameter values to fluctuate very easily, whilst lower values may inhibit learning capabilities. Similarly, altering the regularization coefficient for a Support Vector Machine (SVM) may greatly affect the shape of its decision boundary.
+
+
+\section{Approaches for Hyperparameter Optimization}
+Owing to the necessity of optimal hyperparameter values in machine learning problems, various works of literature are dedicated solely to their optimization. During the past decade, methods have been refined and applied in a large number of model methods. The sections below will go through a number of these approaches found in published works.
+\subsection{Gradient-Based Hyperparameter Optimization}
+ A particular project by  Sathiya \textit{et al.}\cite{SathiyaKeerthi2006} investigates hyperparameter values in SVMs and Bayesian models. In this system, the variables $C$ and $\gamma$ are considered. C is the regularization coefficient, influencing how leniently the SVM hyperplane classifies observations. $\gamma$ defines the influence of points on the variance of a model.  Experiments in this study showed that performance of the SVM varied considerably as a function of hyperparameter values. Gradient-based optimization techniques were used, computing the slope of a performance validation function with respect to the vector $h$ of hyperparameters. The gradient-based system described allowed for a fast method to determine optimal sets of hyperparameters, which outperformed a \textit{grid search} method even for models with solely 2 hyperparameters.
```

Comment:
Be consistent, use $C$ everywhere.  2 should be 'two' in this context.  Reference to grid search but not defined anywhere.